### ISSUE TITLE
User suggestion: electrocute the dog

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/circuit-construction-kit-common#512]
> p.s. while playing, I tried to electrocute the dog and the “hand” taken from the grab bag.  Even at 100000v dc and max ac - the current in dog was 0 and negligible in the “hand”  ... not sure why these examples were given…  maybe a “wet dog” would catch on fire – kinda like the pencil burst into flame - big time!!

**Comment by samreid** [comment]
It seems this would be good to discuss at an upcoming CCK design meeting, particularly if we want to work on this before AC publication.  Also, even if we decide there will be no grab bag items in the AC sim, it may be beneficial to republish DC at the same time as AC to keep them in sync, if time allows.

**Comment by samreid** [comment]
Notes from today's meeting.  Cut the front paw junction when current level exceeded.  Show dog turns head to side, and a short bark sound.

**Comment by arouinfar** [comment]
1/9/20 design meeting notes

* Update dog resistance to match the hand (100000 ohms)
* Get additional artwork of the dog, head turned to face user with mouth slightly open.
* Treat dog like a fuse. When match current reached (0.1 amps), swap out artwork and play "yip" sound.
* Cut junction at the dog's paw. The mechanics would be identical to using the cut button, severing the connection of all components at the junction.
![image](https://user-images.githubusercontent.com/8419308/72115012-b6ea9700-3302-11ea-91b4-b28c46f3f881.png)

**Comment by arouinfar** [comment]
Assigning @samreid to work on the fuse-like mechanics, but leaving self assigned to work with the graphic artist.

**Comment by samreid** [comment]
After the above commits, when the current through the dog exceeds the threshold, it barks and splits the junction at the front paws.  I used placeholders for the bark image and audio, and it should be ready to test functionality.  Self-unassigning until we have improved bark image and audio or feedback on this draft.

**Comment by arouinfar** [comment]
@samreid the behavior is looking good! I see the "Bark" text, but I don't actually hear a sound. Not sure if you have a stand-in audio file, or you just have the architecture ready to go once there is a sound to incorporate.

**Comment by samreid** [comment]
Thanks for letting me know! I had failed to push the *.mp3 file.  After pushing, I confirmed I can hear the audio on phettest.

**Comment by arouinfar** [comment]
Thanks @samreid. Looks like the mechanics are all in place now.

**Comment by samreid** [comment]
On hold until artwork is complete.

**Comment by samreid** [comment]
Increasing priority since this was requested for upcoming release.

**Comment by samreid** [comment]
We reviewed the case in https://github.com/phetsims/circuit-construction-kit-common/issues/512#issuecomment-642244320 and agreed as long as the current is only through the hand and the current is low, it would not be a health risk.

We reviewed the behavior in master and agreed it is OK. We feel there are pedagogical advantages of showing a disembodied hand, so students can see that hands are not perfect insulators.  We don't want to have the hand recoil.

**Comment by samreid** [comment]
We also agreed we would like to have a "dog looks at the camera/head tilt" graphic for the upcoming publication.

**Comment by samreid** [comment]
@arouinfar can you please comment on the expected timeline for this issue?

**Comment by arouinfar** [comment]
The artist who created the dog currently in CCK is on hiatus from PhET until February. I sent an email to @samreid @kathy-phet @ariel-phet to discuss next steps.

**Comment by arouinfar** [comment]
Possible options

1. Work with a different artist, which would probably require new altogether new dog artwork.
2. Wait on adding new artwork until the original artist is available in February. Proceed with publication without the secondary dog artwork, but plan to republish CCK: DC after CCK: AC is published. 
3. Forget secondary dog artwork, and find some other way to visually indicate that the dog barked or the current was too high. Some rough sketches:
![image](https://user-images.githubusercontent.com/8419308/100245628-80574400-2ef5-11eb-9907-b598ec59efa8.png)

**Comment by samreid** [comment]
The cartoonish "!!!" speech bubble looks great to me for next release.

**Comment by kathy-phet** [comment]
I like the cartoon "!!!" as well.

**Comment by ariel-phet** [comment]
Looks good

**Comment by arouinfar** [comment]
I cleaned up the speech bubble a bit. Here are the assets @samreid:
[dog_bark.zip](https://github.com/phetsims/circuit-construction-kit-common/files/5618765/dog_bark.zip)


![dog_bark](https://user-images.githubusercontent.com/8419308/100667256-7f6c4b00-3317-11eb-8640-d8286147d2c3.png)
If you prefer to draw it in code, the speech bubble is fontawesome [f075](https://fontawesome.com/icons/comment?style=regular). The exclamation points are Times New Roman. The right/left exclamation points are 20% smaller and rotated 20 degrees away from the central `!`. You could also go with the fontawesome exclamation point [f12a](https://fontawesome.com/icons/exclamation).

**Comment by samreid** [comment]
We already had the "comment" path data.  I added the exclamation point path, and wired it up in the commit.  It looks like this:

![image](https://user-images.githubusercontent.com/679486/100913860-8ddd7280-348f-11eb-98f7-bedd70c3662b.png)

@arouinfar can you please review in master?

**Comment by arouinfar** [comment]
@samreid the speech bubble is looking great, but its duration feels a bit too short. If you aren't looking at the dog when the current threshold is exceeded, it's easy to miss. 

![bark](https://user-images.githubusercontent.com/8419308/100943004-98acfd00-34b9-11eb-95dd-e73740c6a9f3.gif)

Can we try something like 0.5 s? I realize this means the speech bubble will be visible for a bit longer than the audio duration, but I think that's ok.

**Comment by samreid** [comment]
I increased the time to 0.5 seconds and positioned the bark relative to the dog so it would look OK at different rotations.  Ready for review.

**Comment by samreid** [comment]
Also, I kept the bark always "upside up" like so, but we could rotate it with the dog if we prefer:

![image](https://user-images.githubusercontent.com/679486/101221644-5284a480-3645-11eb-8edc-ccc9c41dfb83.png)

**Comment by arouinfar** [comment]
@samreid the speech bubble duration looks great! In general, I like the "upside up" approach, but there are a few rotations where the dog ends up occluded so it might not be the best solution.
<img width="424" alt="dog1" src="https://user-images.githubusercontent.com/8419308/101395759-a088ea80-3887-11eb-9869-61d9e49647c1.png">
<img width="249" alt="dog2" src="https://user-images.githubusercontent.com/8419308/101395762-a1ba1780-3887-11eb-955a-8674add0f8a0.png">

**Comment by samreid** [comment]
After the commit, the speech bubble rotates with the dog.  I also noticed that electrons or other circuit elements can overlap the speech bubble, but I don't think that would be trivial to change the z-ordering.  @arouinfar ready for review.

**Comment by arouinfar** [comment]
Looking good in master @samreid. I don't see any reason to mess with the z-ordering.

**Comment by kathy-phet** [comment]
@samreid - The dog can still be occluded -- is this expected? 
![image](https://user-images.githubusercontent.com/5096804/102285127-d581e680-3ef2-11eb-8cdd-dd7cb19b8e3f.png)

Also, I noticed that he doesn't yelp until he has a 10K voltage across him - is that expected? I would think he would be electrocuted (as in die) at a lower voltage?

**Comment by kathy-phet** [comment]
I would think a reaction of a yelp at around 110 or 120 volt would be appropriate?

**Comment by samreid** [comment]
@arouinfar and @kathy-phet and I set the threshold to Voltage of 100 V instead of a current of 0.1 Amps.  Leaving open as a reminder to notify the client after this is published.

**Comment by samreid** [comment]
This has been published, @oliver-phet can you please notify the client?  Close when complete.

**Comment by oliver-phet** [comment]
Ok, I've emailed the 3 users about the changes!
